doc/contribute/coding_guidelines/index.rst

```diff
@@ -1297,6 +1297,188 @@ macros. Given that Zephyr uses a fork of the corresponding upstream for each
 module, it is always possible to patch the macro implementation in each module
 to avoid collisions.
 
+.. _coding_guideline_libc_usage_restrictions_in_zephyr_kernel:
+
+Rule A.4: C Standard Library Usage Restrictions in Zephyr Kernel
+================================================================
+
+Severity
+--------
+
+Required
+
+Description
+-----------
+
+The use of the C standard library functions and macros in the Zephyr kernel
+shall be limited to the following functions and macros from the ISO/IEC
+9899:2011 standard, also known as C11, and their extensions:
```

Comment:
Regarding ISO/IEC 9899:2011 standard.  This standard has been [withdrawn](https://www.iso.org/standard/57853.html) and replaced by ISO/IEC 9899:2018.

Do we need a doc scrub as a separate PR to replace 9899:2011 references with 9899:2018?

Reply:
I think we're free to reference the older spec if we please; it's still well defined, even if not current. We should consider when a transition to :2018 is to be made.

Reply:
The reference to the ISO/IEC 9899:2011 (aka. C11) is intentional since the project is still adhering to the ISO/IEC 9899:1999 (aka. C99) for the most part and, at most, C11 (see https://github.com/zephyrproject-rtos/zephyr/issues/30105).

While C17 is technically only bug fixes for the C11, it is still a different version and some toolchains that the Zephyr project and its downstream users depend on may not have the support for it.